WcaOnRails/app/models/competition.rb

```diff
@@ -424,8 +425,10 @@ def belongs_to_region?(region)
   end
 
   def contains?(search_param)
-    [name, cityName, venue, cellName, countryId, start_date.strftime('%B')].any? do |field|
-      field.downcase.include?(search_param.downcase)
+    search_param.split.any? do |part|
+      [name, cityName, delegates.pluck(&:name).join(','), venue, cellName, countryId, start_date.strftime('%B')].any? do |field|
```

Comment:
Nit: could we flip the loops here? I think it's kind of silly to repeatedly run `delegates.pluck(&:name).join(',')`.